pkg/controller/elasticache/user/setup.go

```diff
@@ -0,0 +1,185 @@
+			e.isUpToDate = c.isUpToDate
+			e.preCreate = c.preCreate
+			e.postCreate = postCreate
+			e.preObserve = preObserve
+			e.preDelete = preDelete
+			e.postObserve = postObserve
+			e.postDelete = postDelete
+			e.preUpdate = c.preUpdate
+			e.postUpdate = c.postUpdate
+			e.filterList = filterList
+		},
+	}
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewController(rl),
+		}).
+		For(&svcapitypes.User{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(svcapitypes.UserGroupVersionKind),
+			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient())),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), opts: opts}),
+			managed.WithPollInterval(poll),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type custom struct {
+	kube     client.Client
+	client   svcsdkapi.ElastiCacheAPI
+	external *external
+}
+
+func filterList(cr *svcapitypes.User, obj *svcsdk.DescribeUsersOutput) *svcsdk.DescribeUsersOutput {
+	userIdentifier := awsclients.String(meta.GetExternalName(cr))
+	resp := &svcsdk.DescribeUsersOutput{}
+	for _, user := range obj.Users {
+		if awsclients.StringValue(user.UserId) == awsclients.StringValue(userIdentifier) {
+			resp.Users = append(resp.Users, user)
+			break
+		}
+	}
+	return resp
+}
+
+func (e *custom) preCreate(ctx context.Context, cr *svcapitypes.User, obj *svcsdk.CreateUserInput) error {
+	obj.UserId = awsclients.String(cr.Name)
+	obj.UserName = awsclients.String(cr.Name)
+
+	pw, _, err := elasticache.GetPassword(ctx, e.kube, &cr.Spec.ForProvider.PasswordSecretRef, cr.Spec.WriteConnectionSecretToReference)
+	if resource.IgnoreNotFound(err) != nil {
+		return errors.Wrap(err, "cannot get password from the given secret")
+	}
+
+	if pw != "" {
+		obj.NoPasswordRequired = awsclients.Bool(false)
+		obj.Passwords = append(obj.Passwords, awsclients.String(pw))
+	} else {
+		obj.NoPasswordRequired = awsclients.Bool(true)
+	}
+
+	return nil
+}
+
+func postCreate(_ context.Context, cr *svcapitypes.User, obj *svcsdk.CreateUserOutput, cre managed.ExternalCreation, err error) (managed.ExternalCreation, error) {
+	if err != nil {
+		return managed.ExternalCreation{}, err
+	}
+	meta.SetExternalName(cr, awsclients.StringValue(&cr.Name))
+	return cre, nil
+}
+
+func preObserve(_ context.Context, cr *svcapitypes.User, obj *svcsdk.DescribeUsersInput) error {
+	obj.UserId = awsclients.String(meta.GetExternalName(cr))
+	return nil
+}
+
+func postObserve(_ context.Context, cr *svcapitypes.User, obj *svcsdk.DescribeUsersOutput, obs managed.ExternalObservation, err error) (managed.ExternalObservation, error) {
+	if err != nil {
+		return managed.ExternalObservation{}, err
+	}
+
+	if obj.Users[0] == nil {
+		return managed.ExternalObservation{}, errors.New("no user info found in response")
+	}
```

Comment:
```suggestion
```
If the code comes to that point, we're sure that the user exists because we already did the filtering.

Reply:
done